## B-vac A Robust Software Package for Bacterial Vaccine Design

B-vac is an executable pipeline designed to streamline bacterial vaccine discovery using reverse vaccinology (RV). It leverages bioinformatics to analyze pathogen genomic data, identifying proteins with key traits such as immunogenicity, surface localization, and conservation across strains. B-vac is a user-friendly, offline solution that addresses challenges in prediction accuracy, computational demands, and accessibility faced by existing RV tools.

## Features

1. **User-Friendly Interface:** Simplifies the vaccine design process for researchers.
2. **Offline Operation:** Eliminates dependency on external libraries or internet connectivity.
3. **High-Throughput Proteomics Analysis:** Includes modules for Localization, Non-host Homolog Identification, Virulence Factor Detection and Epitope Mapping.
4. **Customizable Parameters:** Allows users to set variables and toggles for cut-off and filter values.
5. **Curated Databases:** Integrates epitope libraries, bacterial proteomes, and virulence factor databases for local processing.
6. **String-Based Matching:** Aligns and compares pathogen protein sequences to identify immunogenic motifs.
7. **Standalone Solution:** Operates without external dependencies, enhancing accessibility for researchers with limited computational resources.

## Installation
Setup files can be downloaded from https://shorturl.at/GzvvV. 

**How to install - for Linux:**
In terminal run this command after copying the deb package file in home directory:
dpkg-deb --build Bvac_deb

How to run:
After installation, run the following command in terminal:
B-vac

**How to install - for Windows:**
Double click the exe and proceed with windows setup installer.

## License and Copyright

B-vac setups installation and usage is licensed under the MIT License. Usage of source code requires permission. Contact corresponding author: amjad.ali@asab.nust.edu.pk

## Citation

Manuscript accepted in Scientific Reports. Citation will be available soon. 

## Contact

Contact corresponding author: amjad.ali@asab.nust.edu.pk


